light-client/src/types.rs

```diff
@@ -49,30 +49,29 @@ pub type SignedHeader = TMSignedHeader;
 pub type TrustedState = LightBlock;
 
 /// Verification status of a light block.
-#[derive(Copy, Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
 pub enum Status {
+    /// The light block has failed verification.
+    Failed,
     /// The light has not been verified yet.
     Unverified,
     /// The light block has been successfully verified.
     Verified,
     /// The light block has been successfully verified and has passed fork detection.
     Trusted,
-    /// The light block has failed verification.
-    Failed,
 }
 
 impl Status {
     /// Return a slice of all the possible values for this enum.
-    pub fn iter() -> &'static [Status] {
-        static ALL: &[Status] = &[
-            Status::Unverified,
-            Status::Verified,
-            Status::Trusted,
-            Status::Failed,
-        ];
-
+    pub fn iter() -> &'static [Self] {
```

Comment:
Where are we in need of iterating over all statuses?

Reply:
https://github.com/informalsystems/tendermint-rs/blob/9631ead98bc55b0189462450aba7fbc9e2754833/light-client/src/store/sled.rs#L55

Reply:
Thanks for the pointer.